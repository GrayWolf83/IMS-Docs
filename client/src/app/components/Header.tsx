import React from 'react'
import {
	Flex,
	Box,
	Container,
	Heading,
	useColorMode,
	IconButton,
} from '@chakra-ui/react'
import { FcSettings, FcManager } from 'react-icons/fc'
import { FiLogOut } from 'react-icons/fi'
import { Link } from 'react-router-dom'
import ColorModeSwitcher from './ColorModeSwitcher'

const Header: React.FC = () => {
	const { colorMode } = useColorMode()
	const isLogin = true
	const isManage = true

	return (
		<Box
			as='header'
			py='3'
			bg={colorMode === 'dark' ? 'gray.600' : 'gray.300'}>
			<Container maxW='container.xl'>
				<Flex justifyContent='space-between'>
					<Link to='/'>
						<Heading>e-IMS</Heading>
					</Link>

					<Flex alignItems='center'>
						<ColorModeSwitcher />
						{isManage && (
							<Link to='/manage'>
								<IconButton
									size='md'
									fontSize='lg'
									variant='outline'
									color='current'
									marginLeft='2'
									aria-label=''
									icon={<FcSettings fontSize='24px' />}
								/>
							</Link>
						)}
						{isLogin && (
							<>
								<IconButton
									size='md'
									fontSize='lg'
									variant='outline'
									color='current'
									marginLeft='2'
									aria-label=''
									icon={<FcManager fontSize='24px' />}
								/>
								<IconButton
									size='md'
									fontSize='lg'
									variant='outline'
									color='current'
									marginLeft='2'
									aria-label=''
									icon={<FiLogOut fontSize='24px' />}
								/>
							</>
						)}
					</Flex>
				</Flex>
			</Container>
		</Box>
	)
}

export default Header
