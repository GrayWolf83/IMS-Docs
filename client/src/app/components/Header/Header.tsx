import React from 'react'
import {
	Flex,
	Box,
	Container,
	Heading,
	useColorMode,
	IconButton,
} from '@chakra-ui/react'
import { FcManager } from 'react-icons/fc'
import { FiLogOut } from 'react-icons/fi'
import { Link } from 'react-router-dom'
import ColorModeSwitcher from '../ColorModeSwitcher'

const Header: React.FC = () => {
	const { colorMode } = useColorMode()
	const isLogin = true

	return (
		<Box
			data-testid='Header'
			as='header'
			py='3'
			bg={colorMode === 'dark' ? 'gray.600' : 'gray.300'}>
			<Container maxW='container.xl'>
				<Flex justifyContent='space-between'>
					<Link to='/'>
						<Heading>e-IMS</Heading>
					</Link>

					<Flex alignItems='center'>
						<ColorModeSwitcher />
						{isLogin && (
							<>
								<IconButton
									size='md'
									fontSize='lg'
									variant='outline'
									color='current'
									marginLeft='2'
									aria-label=''
									icon={<FcManager fontSize='24px' />}
								/>
								<IconButton
									size='md'
									fontSize='lg'
									variant='outline'
									color='current'
									marginLeft='2'
									aria-label=''
									icon={<FiLogOut fontSize='24px' />}
								/>
							</>
						)}
					</Flex>
				</Flex>
			</Container>
		</Box>
	)
}

export default Header
