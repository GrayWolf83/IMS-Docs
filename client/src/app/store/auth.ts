import { setLoadingError } from './error'
import { IUser } from './../../../../server/types/User'
import { AppDispatch, RootState } from './index'
import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import authService from '../services/auth.service'
import localStorageService from '../services/localStorage.service'

type authState = {
	currentUser: IUser | null
	isLoading: boolean
}

const initialState: authState = {
	currentUser: null,
	isLoading: false,
}

const authSlice = createSlice({
	name: 'auth',
	initialState,
	reducers: {
		authLoginSuccess(state, action: PayloadAction<IUser>) {
			state.currentUser = action.payload
		},
		authLogout(state) {
			state.currentUser = null
		},
		authLoadingStart(state) {
			state.isLoading = true
		},
		authLoadingEnd(state) {
			state.isLoading = false
		},
	},
})

const { authLoginSuccess, authLoadingStart, authLoadingEnd, authLogout } =
	authSlice.actions

export const login = (data: FormData) => async (dispatch: AppDispatch) => {
	dispatch(authLoadingStart())

	try {
		const payload = await authService.login(data)
		dispatch(authLoginSuccess(payload.user))

		localStorageService.setTokens({
			...payload,
		})
	} catch (error) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(authLoadingEnd())
	}
}

export const autologin = () => async (dispatch: AppDispatch) => {
	dispatch(authLoadingStart())
	try {
		const payload = await authService.refresh()

		if (!payload?.refreshToken || !payload?.accessToken) {
			localStorageService.removeAuthData()
		} else {
			dispatch(authLoginSuccess(payload.user))

			localStorageService.setTokens({
				...payload,
			})
		}
	} catch (error: any) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(authLoadingEnd())
	}
}

export const logout = () => async (dispatch: AppDispatch) => {
	localStorageService.removeAuthData()
	dispatch(authLogout())
}

export const getCurrentUser = () => (state: RootState) => {
	return state.auth.currentUser
}

export const getAuthLoadingStatus = () => (state: RootState) => {
	return state.auth.isLoading
}

export const getCurrentUserIsManage = () => (state: RootState) => {
	return Boolean(state.auth.currentUser?.role === 'Manage')
}

export default authSlice.reducer
