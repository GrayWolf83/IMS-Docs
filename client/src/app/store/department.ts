import { setLoadingError } from './error'
import { AppDispatch, RootState } from './index'
import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { IDepartment } from '../types/department'
import departmentService from '../services/department.service'

type state = {
	entities: IDepartment[]
	isLoading: boolean
}

const initialState: state = {
	entities: [],
	isLoading: false,
}

const departmentSlice = createSlice({
	name: 'departments',
	initialState,
	reducers: {
		departmentsLoaded(state, action: PayloadAction<IDepartment[]>) {
			state.entities = action.payload
		},
		departmentAdded(state, action: PayloadAction<IDepartment>) {
			state.entities = [action.payload, ...state.entities]
		},
		departmentEdited(state, action: PayloadAction<IDepartment>) {
			state.entities = state.entities.map((item) => {
				return item.id === action.payload.id ? action.payload : item
			})
		},
		departmentDeleted(state, action: PayloadAction<string>) {
			state.entities = state.entities.filter(
				(item) => item.id !== action.payload,
			)
		},
		departmentLoadingStart(state) {
			state.isLoading = true
		},
		departmentLoadingEnd(state) {
			state.isLoading = false
		},
	},
})

const {
	departmentsLoaded,
	departmentAdded,
	departmentEdited,
	departmentDeleted,
	departmentLoadingStart,
	departmentLoadingEnd,
} = departmentSlice.actions

export const loadDepartmentList = () => async (dispatch: AppDispatch) => {
	dispatch(departmentLoadingStart())

	try {
		const payload = await departmentService.getList()
		dispatch(departmentsLoaded(payload))
	} catch (error) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(departmentLoadingEnd())
	}
}

export const addDepartment =
	(data: { name: string; fullName: string; index: number }) =>
	async (dispatch: AppDispatch) => {
		dispatch(departmentLoadingStart())
		try {
			const payload: IDepartment = await departmentService.add(data)

			dispatch(departmentAdded(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(departmentLoadingEnd())
		}
	}

export const editDepartment =
	(data: IDepartment) => async (dispatch: AppDispatch) => {
		dispatch(departmentLoadingStart())
		try {
			const payload: IDepartment = await departmentService.edit(data)

			dispatch(departmentEdited(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(departmentLoadingEnd())
		}
	}

export const deleteDepartment =
	(data: string) => async (dispatch: AppDispatch) => {
		dispatch(departmentLoadingStart())
		try {
			const payload: string = await departmentService.delete(data)

			dispatch(departmentDeleted(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(departmentLoadingEnd())
		}
	}

export const getDepartmentsList = () => (state: RootState) => {
	return state.departments.entities
		.filter((item) => item.index)
		.sort((a, b) => (a.index > b.index ? 1 : -1))
}

export const getDepartmentById = (id: string) => (state: RootState) => {
	return state.departments.entities.find((item) => item.id === id)
}

export default departmentSlice.reducer
