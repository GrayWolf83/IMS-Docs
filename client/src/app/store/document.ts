import { setLoadingError } from './error'
import { AppDispatch, RootState } from './index'
import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { IDocument, INewDocument } from '../types/document'
import documentService from '../services/document.service'

type state = {
	entities: IDocument[]
	isLoading: boolean
}

const initialState: state = {
	entities: [],
	isLoading: false,
}

const documentSlice = createSlice({
	name: 'documents',
	initialState,
	reducers: {
		loaded(state, action: PayloadAction<IDocument[]>) {
			state.entities = action.payload
		},
		added(state, action: PayloadAction<IDocument>) {
			state.entities = [action.payload, ...state.entities]
		},
		edited(state, action: PayloadAction<IDocument>) {
			state.entities = state.entities.map((item) => {
				return item.id === action.payload.id ? action.payload : item
			})
		},
		deleted(state, action: PayloadAction<string>) {
			state.entities = state.entities.filter(
				(item) => item.id !== action.payload,
			)
		},
		loadingStart(state) {
			state.isLoading = true
		},
		loadingEnd(state) {
			state.isLoading = false
		},
	},
})

const { loaded, added, edited, deleted, loadingStart, loadingEnd } =
	documentSlice.actions

export const loadDocumentsList =
	(systemId: string) => async (dispatch: AppDispatch) => {
		dispatch(loadingStart())

		try {
			const payload: IDocument[] = await documentService.getList(systemId)
			dispatch(loaded(payload))
		} catch (error) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(loadingEnd())
		}
	}

export const addDocument =
	(data: INewDocument) => async (dispatch: AppDispatch) => {
		dispatch(loadingStart())
		try {
			const payload: IDocument = await documentService.add(data)

			dispatch(added(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(loadingEnd())
		}
	}

export const editDocument =
	(data: IDocument) => async (dispatch: AppDispatch) => {
		dispatch(loadingStart())
		try {
			const payload: IDocument = await documentService.edit(data)

			dispatch(edited(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(loadingEnd())
		}
	}

export const deleteDocument =
	(data: string) => async (dispatch: AppDispatch) => {
		dispatch(loadingStart())
		try {
			const payload: string = await documentService.delete(data)

			dispatch(deleted(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(loadingEnd())
		}
	}

export const getDocumentsList = () => (state: RootState) => {
	return state.documents.entities
}

export const getDocumentById = (id: string) => (state: RootState) => {
	return state.documents.entities.find((item) => item.id === id)
}

export default documentSlice.reducer
