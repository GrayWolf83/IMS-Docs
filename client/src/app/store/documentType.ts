import { setLoadingError } from './error'
import { AppDispatch, RootState } from './index'
import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { IDocumentType, INewDocumentType } from '../types/documentType'
import documentTypeService from '../services/documentType.service'

type state = {
	entities: IDocumentType[]
	isLoading: boolean
}

const initialState: state = {
	entities: [],
	isLoading: false,
}

const documentTypeSlice = createSlice({
	name: 'documentTypes',
	initialState,
	reducers: {
		documentTypesLoaded(state, action: PayloadAction<IDocumentType[]>) {
			state.entities = action.payload
		},
		documentTypeAdded(state, action: PayloadAction<IDocumentType>) {
			state.entities = [action.payload, ...state.entities]
		},
		documentTypeEdited(state, action: PayloadAction<IDocumentType>) {
			state.entities = state.entities.map((item) => {
				return item.id === action.payload.id ? action.payload : item
			})
		},
		documentTypeDeleted(state, action: PayloadAction<string>) {
			state.entities = state.entities.filter(
				(item) => item.id !== action.payload,
			)
		},
		documentTypeLoadingStart(state) {
			state.isLoading = true
		},
		documentTypeLoadingEnd(state) {
			state.isLoading = false
		},
	},
})

const {
	documentTypesLoaded,
	documentTypeAdded,
	documentTypeEdited,
	documentTypeDeleted,
	documentTypeLoadingStart,
	documentTypeLoadingEnd,
} = documentTypeSlice.actions

export const loadDocumentTypesList = () => async (dispatch: AppDispatch) => {
	dispatch(documentTypeLoadingStart())

	try {
		const payload: IDocumentType[] = await documentTypeService.getList()
		dispatch(documentTypesLoaded(payload))
	} catch (error) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(documentTypeLoadingEnd())
	}
}

export const addDocumentType =
	(data: INewDocumentType) => async (dispatch: AppDispatch) => {
		dispatch(documentTypeLoadingStart())
		try {
			const payload: IDocumentType = await documentTypeService.add(data)

			dispatch(documentTypeAdded(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(documentTypeLoadingEnd())
		}
	}

export const editDocumentType =
	(data: IDocumentType) => async (dispatch: AppDispatch) => {
		dispatch(documentTypeLoadingStart())
		try {
			const payload: IDocumentType = await documentTypeService.edit(data)

			dispatch(documentTypeEdited(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(documentTypeLoadingEnd())
		}
	}

export const deleteDocumentType =
	(data: string) => async (dispatch: AppDispatch) => {
		dispatch(documentTypeLoadingStart())
		try {
			const payload: string = await documentTypeService.delete(data)

			dispatch(documentTypeDeleted(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(documentTypeLoadingEnd())
		}
	}

export const getDocumentTypesList = () => (state: RootState) => {
	return state.documentTypes.entities
}

export const getDocumentTypeById = (id: string) => (state: RootState) => {
	return state.documentTypes.entities.find((item) => item.id === id)
}

export default documentTypeSlice.reducer
