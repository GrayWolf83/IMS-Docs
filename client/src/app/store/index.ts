import { configureStore, combineReducers } from '@reduxjs/toolkit'
import errorReducer from './error'

const store = configureStore({
	reducer: combineReducers({
		errors: errorReducer,
	}),
})

export default store

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
