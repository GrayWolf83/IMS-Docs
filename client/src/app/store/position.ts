import { setLoadingError } from './error'
import { AppDispatch, RootState } from './index'
import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { IPosition } from '../types/position'
import positionService from '../services/position.service'

type state = {
	entities: IPosition[]
	isLoading: boolean
}

const initialState: state = {
	entities: [],
	isLoading: false,
}

const positionSlice = createSlice({
	name: 'positions',
	initialState,
	reducers: {
		positionsLoaded(state, action: PayloadAction<IPosition[]>) {
			state.entities = action.payload
		},
		positionAdded(state, action: PayloadAction<IPosition>) {
			state.entities = [action.payload, ...state.entities]
		},
		positionEdited(state, action: PayloadAction<IPosition>) {
			state.entities = state.entities.map((item) => {
				return item.id === action.payload.id ? action.payload : item
			})
		},
		positionDeleted(state, action: PayloadAction<string>) {
			state.entities = state.entities.filter(
				(item) => item.id !== action.payload,
			)
		},
		positionLoadingStart(state) {
			state.isLoading = true
		},
		positionLoadingEnd(state) {
			state.isLoading = false
		},
	},
})

const {
	positionsLoaded,
	positionAdded,
	positionEdited,
	positionDeleted,
	positionLoadingStart,
	positionLoadingEnd,
} = positionSlice.actions

export const loadPositionsList = () => async (dispatch: AppDispatch) => {
	dispatch(positionLoadingStart())

	try {
		const payload: IPosition[] = await positionService.getList()
		dispatch(positionsLoaded(payload))
	} catch (error) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(positionLoadingEnd())
	}
}

export const addPosition =
	(data: { name: string; index: number }) =>
	async (dispatch: AppDispatch) => {
		dispatch(positionLoadingStart())
		try {
			const payload: IPosition = await positionService.add(data)

			dispatch(positionAdded(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(positionLoadingEnd())
		}
	}

export const editPosition =
	(data: IPosition) => async (dispatch: AppDispatch) => {
		dispatch(positionLoadingStart())
		try {
			const payload: IPosition = await positionService.edit(data)

			dispatch(positionEdited(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(positionLoadingEnd())
		}
	}

export const deletePosition =
	(data: string) => async (dispatch: AppDispatch) => {
		dispatch(positionLoadingStart())
		try {
			const payload: string = await positionService.delete(data)

			dispatch(positionDeleted(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(positionLoadingEnd())
		}
	}

export const getPositionsList = () => (state: RootState) => {
	return state.positions.entities
		.filter((item) => item.index)
		.sort((a, b) => (a.index > b.index ? 1 : -1))
}

export const getPositionById = (id: string) => (state: RootState) => {
	return state.positions.entities.find((item) => item.id === id)
}

export default positionSlice.reducer
