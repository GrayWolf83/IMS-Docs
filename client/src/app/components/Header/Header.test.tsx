import React from 'react'
import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import store from '../../store'
import Header from './Header'

describe('Header component', () => {
	test('Header render', () => {
		render(
			<Provider store={store}>
				<BrowserRouter>
					<Header />
				</BrowserRouter>
			</Provider>,
		)

		const header = screen.getByTestId('Header')
		expect(header).toBeInTheDocument()
	})

	test('Header snapshot', () => {
		expect(
			render(
				<Provider store={store}>
					<BrowserRouter>
						<Header />
					</BrowserRouter>
				</Provider>,
			),
		).toMatchSnapshot()
	})
})
