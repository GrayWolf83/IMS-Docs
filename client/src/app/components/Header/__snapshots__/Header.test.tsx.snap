// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="css-0"
        data-testid="Header"
      >
        <div
          class="chakra-container css-0"
        >
          <div
            class="css-0"
          >
            <a
              href="/"
            >
              <h2
                class="chakra-heading css-0"
              >
                e-IMS
              </h2>
            </a>
            <div
              class="css-0"
            >
              <button
                aria-label="Switch to dark mode"
                class="chakra-button css-0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
                  />
                </svg>
              </button>
              <a
                href="/manage"
              >
                <button
                  aria-label=""
                  class="chakra-button css-0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    enable-background="new 0 0 48 48"
                    fill="currentColor"
                    focusable="false"
                    font-size="24px"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    version="1"
                    viewBox="0 0 48 48"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M39.6,27.2c0.1-0.7,0.2-1.4,0.2-2.2s-0.1-1.5-0.2-2.2l4.5-3.2c0.4-0.3,0.6-0.9,0.3-1.4L40,10.8 c-0.3-0.5-0.8-0.7-1.3-0.4l-5,2.3c-1.2-0.9-2.4-1.6-3.8-2.2l-0.5-5.5c-0.1-0.5-0.5-0.9-1-0.9h-8.6c-0.5,0-1,0.4-1,0.9l-0.5,5.5 c-1.4,0.6-2.7,1.3-3.8,2.2l-5-2.3c-0.5-0.2-1.1,0-1.3,0.4l-4.3,7.4c-0.3,0.5-0.1,1.1,0.3,1.4l4.5,3.2c-0.1,0.7-0.2,1.4-0.2,2.2 s0.1,1.5,0.2,2.2L4,30.4c-0.4,0.3-0.6,0.9-0.3,1.4L8,39.2c0.3,0.5,0.8,0.7,1.3,0.4l5-2.3c1.2,0.9,2.4,1.6,3.8,2.2l0.5,5.5 c0.1,0.5,0.5,0.9,1,0.9h8.6c0.5,0,1-0.4,1-0.9l0.5-5.5c1.4-0.6,2.7-1.3,3.8-2.2l5,2.3c0.5,0.2,1.1,0,1.3-0.4l4.3-7.4 c0.3-0.5,0.1-1.1-0.3-1.4L39.6,27.2z M24,35c-5.5,0-10-4.5-10-10c0-5.5,4.5-10,10-10c5.5,0,10,4.5,10,10C34,30.5,29.5,35,24,35z"
                      fill="#607D8B"
                    />
                    <path
                      d="M24,13c-6.6,0-12,5.4-12,12c0,6.6,5.4,12,12,12s12-5.4,12-12C36,18.4,30.6,13,24,13z M24,30 c-2.8,0-5-2.2-5-5c0-2.8,2.2-5,5-5s5,2.2,5,5C29,27.8,26.8,30,24,30z"
                      fill="#455A64"
                    />
                  </svg>
                </button>
              </a>
              <button
                aria-label=""
                class="chakra-button css-0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  enable-background="new 0 0 48 48"
                  fill="currentColor"
                  focusable="false"
                  font-size="24px"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  version="1"
                  viewBox="0 0 48 48"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polygon
                    fill="#FF9800"
                    points="24,37 19,31 19,25 29,25 29,31"
                  />
                  <g
                    fill="#FFA726"
                  >
                    <circle
                      cx="33"
                      cy="19"
                      r="2"
                    />
                    <circle
                      cx="15"
                      cy="19"
                      r="2"
                    />
                  </g>
                  <path
                    d="M33,13c0-7.6-18-5-18,0c0,1.1,0,5.9,0,7c0,5,4,9,9,9s9-4,9-9C33,18.9,33,14.1,33,13z"
                    fill="#FFB74D"
                  />
                  <path
                    d="M24,4c-6.1,0-10,4.9-10,11c0,0.8,0,2.3,0,2.3l2,1.7v-5l12-4l4,4v5l2-1.7c0,0,0-1.5,0-2.3c0-4-1-8-6-9l-1-2 H24z"
                    fill="#FF5722"
                  />
                  <g
                    fill="#784719"
                  >
                    <circle
                      cx="28"
                      cy="19"
                      r="1"
                    />
                    <circle
                      cx="20"
                      cy="19"
                      r="1"
                    />
                  </g>
                  <path
                    d="M29,31L29,31l-5,1l-5-1c0,0-11,2-11,13h32C40,33,29,31,29,31z"
                    fill="#CFD8DC"
                  />
                  <polygon
                    fill="#3F51B5"
                    points="23,35 22,44 26,44 25,35 26,34 24,32 22,34"
                  />
                </svg>
              </button>
              <button
                aria-label=""
                class="chakra-button css-0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="none"
                  focusable="false"
                  font-size="24px"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
                  />
                  <polyline
                    points="16 17 21 12 16 7"
                  />
                  <line
                    x1="21"
                    x2="9"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="css-0"
      data-testid="Header"
    >
      <div
        class="chakra-container css-0"
      >
        <div
          class="css-0"
        >
          <a
            href="/"
          >
            <h2
              class="chakra-heading css-0"
            >
              e-IMS
            </h2>
          </a>
          <div
            class="css-0"
          >
            <button
              aria-label="Switch to dark mode"
              class="chakra-button css-0"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
                />
              </svg>
            </button>
            <a
              href="/manage"
            >
              <button
                aria-label=""
                class="chakra-button css-0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  enable-background="new 0 0 48 48"
                  fill="currentColor"
                  focusable="false"
                  font-size="24px"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  version="1"
                  viewBox="0 0 48 48"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M39.6,27.2c0.1-0.7,0.2-1.4,0.2-2.2s-0.1-1.5-0.2-2.2l4.5-3.2c0.4-0.3,0.6-0.9,0.3-1.4L40,10.8 c-0.3-0.5-0.8-0.7-1.3-0.4l-5,2.3c-1.2-0.9-2.4-1.6-3.8-2.2l-0.5-5.5c-0.1-0.5-0.5-0.9-1-0.9h-8.6c-0.5,0-1,0.4-1,0.9l-0.5,5.5 c-1.4,0.6-2.7,1.3-3.8,2.2l-5-2.3c-0.5-0.2-1.1,0-1.3,0.4l-4.3,7.4c-0.3,0.5-0.1,1.1,0.3,1.4l4.5,3.2c-0.1,0.7-0.2,1.4-0.2,2.2 s0.1,1.5,0.2,2.2L4,30.4c-0.4,0.3-0.6,0.9-0.3,1.4L8,39.2c0.3,0.5,0.8,0.7,1.3,0.4l5-2.3c1.2,0.9,2.4,1.6,3.8,2.2l0.5,5.5 c0.1,0.5,0.5,0.9,1,0.9h8.6c0.5,0,1-0.4,1-0.9l0.5-5.5c1.4-0.6,2.7-1.3,3.8-2.2l5,2.3c0.5,0.2,1.1,0,1.3-0.4l4.3-7.4 c0.3-0.5,0.1-1.1-0.3-1.4L39.6,27.2z M24,35c-5.5,0-10-4.5-10-10c0-5.5,4.5-10,10-10c5.5,0,10,4.5,10,10C34,30.5,29.5,35,24,35z"
                    fill="#607D8B"
                  />
                  <path
                    d="M24,13c-6.6,0-12,5.4-12,12c0,6.6,5.4,12,12,12s12-5.4,12-12C36,18.4,30.6,13,24,13z M24,30 c-2.8,0-5-2.2-5-5c0-2.8,2.2-5,5-5s5,2.2,5,5C29,27.8,26.8,30,24,30z"
                    fill="#455A64"
                  />
                </svg>
              </button>
            </a>
            <button
              aria-label=""
              class="chakra-button css-0"
              type="button"
            >
              <svg
                aria-hidden="true"
                enable-background="new 0 0 48 48"
                fill="currentColor"
                focusable="false"
                font-size="24px"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                version="1"
                viewBox="0 0 48 48"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  fill="#FF9800"
                  points="24,37 19,31 19,25 29,25 29,31"
                />
                <g
                  fill="#FFA726"
                >
                  <circle
                    cx="33"
                    cy="19"
                    r="2"
                  />
                  <circle
                    cx="15"
                    cy="19"
                    r="2"
                  />
                </g>
                <path
                  d="M33,13c0-7.6-18-5-18,0c0,1.1,0,5.9,0,7c0,5,4,9,9,9s9-4,9-9C33,18.9,33,14.1,33,13z"
                  fill="#FFB74D"
                />
                <path
                  d="M24,4c-6.1,0-10,4.9-10,11c0,0.8,0,2.3,0,2.3l2,1.7v-5l12-4l4,4v5l2-1.7c0,0,0-1.5,0-2.3c0-4-1-8-6-9l-1-2 H24z"
                  fill="#FF5722"
                />
                <g
                  fill="#784719"
                >
                  <circle
                    cx="28"
                    cy="19"
                    r="1"
                  />
                  <circle
                    cx="20"
                    cy="19"
                    r="1"
                  />
                </g>
                <path
                  d="M29,31L29,31l-5,1l-5-1c0,0-11,2-11,13h32C40,33,29,31,29,31z"
                  fill="#CFD8DC"
                />
                <polygon
                  fill="#3F51B5"
                  points="23,35 22,44 26,44 25,35 26,34 24,32 22,34"
                />
              </svg>
            </button>
            <button
              aria-label=""
              class="chakra-button css-0"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                font-size="24px"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
                />
                <polyline
                  points="16 17 21 12 16 7"
                />
                <line
                  x1="21"
                  x2="9"
                  y1="12"
                  y2="12"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
